pkg/controller/ps/controller.go

```diff
@@ -1017,15 +1024,22 @@ func (r *PerconaServerMySQLReconciler) reconcileMySQLRouterConfiguration(ctx con
 		return "", errors.Wrapf(err, "get ConfigMap/%s", cmName)
 	}
 
-	// Cleanup if user removed the configuration from CR
 	if cr.Spec.Proxy.Router.Configuration == "" {
 		exists, err := k8s.ObjectExists(ctx, r.Client, nn, currCm)
 		if err != nil {
 			return "", errors.Wrapf(err, "check if ConfigMap/%s exists", cmName)
 		}
 
 		if !exists || !metav1.IsControlledBy(currCm, cr) {
-			return "", nil
+			//ConfigMap exists and is created by the user
```

Comment:
if we're here it doesn't mean "configmap exists and is created by the user". it means cm doesn't exist or it's created by user. do we want to calculate the hash if it doesn't exist?

Reply:
[0c79c50](https://github.com/percona/percona-server-mysql-operator/pull/381/commits/0c79c50f7995943f3e21439e5d3712df04a50184)